### PR TITLE
Unit test for Magento\Reports\Observer\CheckoutCartAddProductObserver

### PR DESCRIPTION
**Pull request by karyna-t**
### Description (*)
Covers Magento\Reports\Observer\CheckoutCartAddProductObserver with a unit test.

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Review by aleron75 — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @aleron75, thank you for the review.
[ENGCOM-6767](https://jira.corp.magento.com/browse/ENGCOM-6767) has been created to process this Pull Request

**Comment by engcom-Bravo** [comment]
:heavy_check_mark: QA Passed

The test is added and passes 
CheckoutCartAddProductObserverTest

File: app/code/Magento/Reports/Test/Unit/Observer/CheckoutCartAddProductObserverTest.php [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+use Magento\Quote\Model\Quote\Item as QuoteItem;
+use Magento\Reports\Model\Event as ReportsEventModel;
+use Magento\Reports\Model\ReportStatus;
+use Magento\Reports\Observer\CheckoutCartAddProductObserver;
+use Magento\Reports\Observer\EventSaver;
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * Unit Test for @see CheckoutCartAddProductObserver
```
Comment (lbajsarowicz) [inline app/code/Magento/Reports/Test/Unit/Observer/CheckoutCartAddProductObserverTest.php]: What's the point of adding @see there? :-)

File: app/code/Magento/Reports/Test/Unit/Observer/CheckoutCartAddProductObserverTest.php [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+                'eventSaver' => $this->eventSaverMock,
+                'reportStatus' => $this->reportStatusMock
+            ]
+        );
+    }
+
+    /**
+     * The case when event has to be successfully saved
+     */
+    public function testExecute()
```
Comment (lbajsarowicz) [inline app/code/Magento/Reports/Test/Unit/Observer/CheckoutCartAddProductObserverTest.php]: Give more meaningful name to test.
`testExecute` is bad practice, as it should explain what we expect to be tested.
For example: `testExecuteExpectsSaveCalledOnceForNewProduct()`

File: app/code/Magento/Reports/Test/Unit/Observer/CheckoutCartAddProductObserverTest.php [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+            ->method('getItem')
+            ->willReturn($this->quoteItemMock);
+    }
+
+    /**
+     * Checking that the method will be never executed
+     */
+    private function checkOriginalMethodIsNeverExecuted()
+    {
+        $this->eventSaverMock->expects($this->never())
```
Comment (lbajsarowicz) [inline app/code/Magento/Reports/Test/Unit/Observer/CheckoutCartAddProductObserverTest.php]: That actually does not exceeds 80 characters in a row, so you can keep that single-line.

**Review by lbajsarowicz — CHANGES_REQUESTED** [review]
@karyna-tsymbal-atwix I love the progress that you've made with Tests!
Just a few hints to your tests and then this will be another example of your perfect work! :+1:

File: app/code/Magento/Reports/Test/Unit/Observer/CheckoutCartAddProductObserverTest.php (hunk 1 above)
Comment (karyna-t) [reply app/code/Magento/Reports/Test/Unit/Observer/CheckoutCartAddProductObserverTest.php]: sorry, it's my old habit :) I've replaced it with a name of the class. Is that ok?

**Review by lbajsarowicz — APPROVED** [review]
Thanks for your contribution

**Comment by magento-engcom-team** [comment]
Hi @lbajsarowicz, thank you for the review.
[ENGCOM-6767](https://jira.corp.magento.com/browse/ENGCOM-6767) has been created to process this Pull Request

**Comment by engcom-Bravo** [comment]
:heavy_check_mark:  QA Passed

**Comment by engcom-Charlie** [comment]
Failures of functional tests, not related to the changes in this PR, test failing on 2.4-develop branch
image
image

**Comment by lbajsarowicz** [comment]
This is actually addressed. Waiting for fixes to be handled.
https://github.com/magento/magento2/pull/26608

**Pull request merged**